### ISSUE TITLE
Searched history

### ISSUE THREAD

**Issue by craigjmidwinter** [mmgoodnow/cross-seed#24]
Thanks for this great project! I'm using this in conjunction with AutoTorrent. I'm currently just running a full search every 6 hours so that I'm not hammering jackett or the indexers, this works well, but I was thinking of ways to increase frequency in order to maximize the potential upload bonus. I was thinking that maybe adding another file to the .cross-seed where the torrents unique identifier as well as a timestamp for the first seen and last search time could open up the possibility to do date-based invocations and running different sync types at different intervals.

The use case would be, if you are scheduling this via cron, you could:
Every minute, only search torrents that have been added since the last run to catch newly grabbed torrents that may already exist on other trackers.

Every X minutes, search torrents recently added that haven't been searched in Y minutes to catch newly grabbed torrents that have just been added on another tracker

And then run a full sync daily or something.

**Comment by mmgoodnow** [comment]
Glad you like it! That's actually part of my next planned feature, #10. Right now the cache is incredibly barebones, and storing a lot more data in the cache will let cross-seed do things like unseen-only searching, etc. I'm not sure when I'll have time to work on it but probably by the end of the year. 

Until then, have you checked out daemon mode? I'm working on improving support for it, and would love to hear suggestions. I'm pretty sure I'm the only user right now so it's very tightly coupled to my own setup.

**Comment by craigjmidwinter** [comment]
Cool! I'll checkout daemon mode this weekend. I'm using deluge, not rutorrent, but maybe I'll take a poke at integrating it this weekend if I have time

**Comment by mmgoodnow** [comment]
Cool! The only thing rtorrent-specific is the function `getInfoHashesToExclude` (at the bottom) which relies on the torrent files being named with their info hashes. if Deluge does this too then it should just work out of the box.

Otherwise, I'm sure a less-performant solution could be easily used instead, something like the following:
```js
function getInfoHashesToExclude() { 
	const { torrentDir } = getRuntimeConfig(); 
	return loadTorrentDir(torrentDir).map((t) => t.infoHash); 
} 
```
It ain't pretty; it'll load the full contents of all of your torrent files into memory temporarily, every time you ping the daemon, which could take a few seconds if you have a lot of torrents. My long term solution for this is to remember the info hashes in the cache, but this should work in a pinch. Good luck!

https://github.com/mmgoodnow/cross-seed/blob/b57412c7481c8cc16852e84afedeb121445d0ffd/src/torrent.js#L39-L44

**Comment by mmgoodnow** [comment]
Closed by #25
